### DISCUSSION TITLE
Debug Reducer, Ignoring Certain Actions

### DISCUSSION THREAD

**Discussion by rehatkathuria** [pointfreeco/swift-composable-architecture#1154] [Ideas]
Building an app that interacts with the AV layer I have a scenario where an action is being fired 60 times a second. Debugging state on this reducer is very hard to do. 

I'd love to see an addition to `.debug()` where I can also pass in an argument of `Action...` to be ignored in the logs

**Comment by iampatbrown** [comment]
Hey @rehatkathuria 👋

You could achieve this using [`.debug(state:action:)`](https://github.com/pointfreeco/swift-composable-architecture/blob/cfcf8b43adc7d521a8ed26992d7ad8b979d386d2/Sources/ComposableArchitecture/Debugging/ReducerDebugging.swift#L99-L102) which takes a `CasePath`. Using `Action...` as an argument would probably only be suitable when the actions are `Equatable` without an associated value. 

Here's an example:

```swift
struct State: Equatable { var value = "Blob" }

enum Action { case foo, bar, baz }

let reducer = Reducer<State, Action, Void>.empty.debug(
  state: { $0 },
  action: CasePath { $0 } extract: { action in
    switch action {
    case .foo, .bar: return action // debug
    case .baz: return nil // ignore
    }
  }
)
```

You could use this approach to create an extension that takes an `Action...` argument as well.

**Comment by rehatkathuria** [comment]
Ah, perfect, cheers @iampatbrown!
